### PR TITLE
Resolve codefactor errors 

### PR DESCRIPTION
**Pull request by prachigarg19**

### CODE REVIEW

File: src/features/projects/components/projects/Filters.tsx [hunk 1]
```diff
@@ -11,6 +11,11 @@ export default function Filters(): ReactElement {
   const { projects, setFilteredProjects, filtersOpen, setFilterOpen } =
     useProjectProps();
 
+  const purpose = {
+    restoration: true,
+    conservation: true,
+  };
+
   const [type, setType] = React.useState<Record<string, boolean>>({
```
Comment (mohitb35) [inline src/features/projects/components/projects/Filters.tsx:19]: Why was this added?

File: src/features/projects/components/ProjectSnippet.tsx [hunk 2]
```diff
@@ -63,7 +63,12 @@ export default function ProjectSnippet({
 
   const { token } = useUserProps();
   const handleOpen = () => {
-    const url = getDonationUrl(project.slug, token, embed, callbackUrl);
+    const url = getDonationUrl(
+      project.slug,
+      token,
+      embed || 'false',
```
Comment (mohitb35) [inline src/features/projects/components/ProjectSnippet.tsx:69]: Any reason you've kept the fallback for `embed` as 'false'? Why not undefined instead?

File: src/features/projects/components/ProjectsMap.tsx [hunk 3]
```diff
@@ -153,7 +153,7 @@ export default function ProjectsMap(): ReactElement {
         {...mapState}
         {...viewport}
         onViewportChange={_onViewportChange}
-        onStateChange={_onStateChange}
+        onViewStateChange={_onStateChange}
```
Comment (mohitb35) [inline src/features/projects/components/ProjectsMap.tsx:156]: The map does not seem to work correctly after this change.

File: src/tenants/common/Home/index.tsx [hunk 4]
```diff
@@ -41,7 +45,7 @@ export default function About({ leaderboard, tenantScore }: Props) {
           {t(`tenants:${config.tenantName}.title`)}
         </p>
 
-        {config.home.descriptionTitle && (
+        {config.home !== undefined && config.home.descriptionTitle && (
           <p
```
Comment (mohitb35) [inline src/tenants/common/Home/index.tsx:49]: There is still a warning here: 

`Property 'descriptionTitle' does not exist on type '{ image: string; }'.`

File: src/tenants/common/Home/index.tsx [hunk 5]
```diff
@@ -6,18 +6,22 @@ import Footer from '../../../features/common/Layout/Footer';
...
-  tenantScore: any;
+  leaderboard: LeaderBoardList | null;
+  tenantScore: TenantScore | null;
 }
 
 export default function About({ leaderboard, tenantScore }: Props) {
   const { t, ready } = useTranslation(['tenants']);
 
-  const descriptionRef = React.useRef(null);
+  const descriptionRef = React.useRef<HTMLElement>(null);
```
Comment (mohitb35) [inline src/tenants/common/Home/index.tsx:24]: `HTMLParagraphELement` probably fits better. There is a warning while using `descriptionRef` currently

File: src/features/projects/components/ProjectsMap.tsx [hunk 6]
```diff
@@ -207,3 +207,7 @@ export default function ProjectsMap(): ReactElement {
     </div>
   );
 }
+interface DetailsType {
+  coordinates: number[];
+  show: boolean;
+}
```
Comment (mohitb35) [inline src/features/projects/components/ProjectsMap.tsx:213]: Is this needed anywhere else?

Comment (prachigarg19) [reply src/features/projects/components/ProjectsMap.tsx:213]: Not that I could find.
